_posts/2020-09-19-github-actions-with-faasd.md

```diff
@@ -0,0 +1,256 @@
+date: 2020-09-19
+image: /images/2020-09-19-github-actions-faasd/lego.jpg
+categories:
+  - faasd
+  - containerd
+  - cicd
+  - github
+author_staff_member: utsav
+dark_background: true
+---
+
+Learn how to bring CI-CD to your OpenFaaS functions running on a faasd appliance with Github Actions
+
+## A few words about faasd
+
+OpenFaaS when came into existence gave the promise of giving users freedom to run their serverless functions anywhere, be it on a Raspberry Pi, in a data center or in the cloud. As of today OpenFaaS offers faas-netes to run OpenFaaS on Kubernetes, which is the recommended way to run OpenFaaS when running your functions in production.
+Though, it's not necessary that you might want or even infact need Kubernetes, especially when considering the complexities involved with managing a Kubernetes cluster, and that's how faasd was born.
+faasd is just like OpenFaaS with the same ecosystem but it spares you from the complexity involved that comes with mangaging a distributed system.
+
+In this tutorial, we'll see how we can get OpenFaaS functions up and running on a faasd appliance with Github Actions.
+
+
+## Prerequisites
+
+* A VM exposed to the Internet
+
+For this tutorial we'll be using a [DigitalOcean](https://digitalocean.com/) Droplet.
+
+* faas-cli
+
+faas-cli is a command line tool used to interact with OpenFaaS, from deploying your functions to viewing logs.
+Get the installation instructions [here](https://docs.openfaas.com/cli/install/)
+
+> You can follow along with the code used in this tutorial from this [repo](https://github.com/utsavanand2/faasd-hello)
+
+## Create a Droplet with faasd
+
+Create a DigitalOcean Droplet with a Ubuntu distribution, with any plan according to what you want to run, faasd being light weight I'm going to use the $5 basic plan.
+
+![Choose a distribution and a plan](/images/2020-09-19-github-actions-faasd/choose-distro.jpg)
+
+Under `Select additional options`, select `User data` and paste in the contents of the cloud-config file from [here](https://github.com/openfaas/faasd/blob/master/cloud-config.txt)
+
+![Select additional options](/images/2020-09-19-github-actions-faasd/additional-options-cloud-config.jpg)
+
+
+![faasd droplet](/images/2020-09-19-github-actions-faasd/faasd-droplet.jpg)
+
+Within as little as 30 seconds, your faasd instance will be up and running.
+
+## Get the OpenFaaS gateway password
+
+Get the gateway password by running the following commands
+
+```sh
+export IP="143.110.184.36"
+export PASSWORD=$(ssh root@$IP "sudo cat /var/lib/faasd/secrets/basic-auth-password")
+```
+
+Assuming that you have faas-cli already installed, login to faasd with faas-cli with the following commands
+
+```sh
+export OPENFAAS_URL=http://$IP:8080
+echo $PASSWORD | faas-cli login --password-stdin -u admin -g $OPENFAAS_URL
+```
+
+## Create an OpenFaaS function
+
+Now that we've logged into faasd with faas-cli, we're ready to deploy functions to it.
```

Comment:
You have the word deploy very close together here, it might read a little better if you don't user "deploy" in the first sentence, for example.
```suggestion
Now that we've logged into faasd with faas-cli, we're ready to run our functions.
```

Reply:
Thank you so much @LucasRoesler for such a thorough review ❤️
I'll make all the suggested changes.